app/helpers/template_helpers.py

```diff
@@ -14,215 +14,299 @@
+            "account_service_log_out_url"
+        )
+        self._cookie_settings_url = current_app.config["COOKIE_SETTINGS_URL"]
+        self._cdn_url = (
+            f'{current_app.config["CDN_URL"]}{current_app.config["CDN_ASSETS_PATH"]}'
+        )
+        self._address_lookup_api = current_app.config["ADDRESS_LOOKUP_API_URL"]
+        self._google_tag_manager_id = current_app.config.get("EQ_GOOGLE_TAG_MANAGER_ID")
+        self._google_tag_manager_auth = current_app.config.get(
+            "EQ_GOOGLE_TAG_MANAGER_AUTH"
         )
 
-    default_context = {
-        "lang": language_code,
-        "crest": True,
-        "newTabWarning": lazy_gettext("The following links open in a new tab"),
-        "copyrightDeclaration": {
-            "copyright": lazy_gettext(
-                "Crown copyright and database rights 2021 NIMA MOU577.501."
-            )
-            if theme == "census-nisra"
-            else lazy_gettext("Crown copyright and database rights 2020 OS 100019153."),
-            "text": lazy_gettext(
-                "Use of address data is subject to the terms and conditions."
+    @property
+    def context(self) -> Dict[str, Any]:
+        return {
+            "account_service_url": self._account_service_url,
+            "account_service_log_out_url": self._account_service_log_out_url,
+            "contact_us_url": asdict(
+                Link("Contact us", f"{self._base_url}contact-us/")
             ),
-        },
-        "rows": [
-            {"itemsList": items_list},
-        ],
-        "legal": [
-            {
-                "itemsList": [
-                    {
-                        "text": lazy_gettext("Cookies"),
-                        "url": static_content_urls["cookies"],
-                        "target": "_blank",
-                    },
-                    {
-                        "text": lazy_gettext("Accessibility statement"),
-                        "url": static_content_urls["accessibility_statement"],
-                        "target": "_blank",
-                    },
-                    {
-                        "text": lazy_gettext("Privacy and data protection"),
-                        "url": static_content_urls["privacy_and_data_protection"],
-                        "target": "_blank",
-                    },
-                    {
-                        "text": lazy_gettext("Terms and conditions"),
-                        "url": static_content_urls["terms_and_conditions"],
-                        "target": "_blank",
-                    },
-                ],
-            },
-        ],
-    }
+            "cookie_settings_url": self._cookie_settings_url,
+            "page_header": self.page_header_context,
+            "footer": self.footer_context,
+            "languages": get_languages_context(),
+            "theme": self._map_theme(self._theme),
+            "language_code": get_locale().language,
```

Comment:
language should be passed in, as we already call `get_locale().language` prior to this. We should also consider passing it to `get_languages_context` as that would mean the language helper also wouldn't need to call `get_locale().language`.

Reply:
Done 0e1ecab1258